### PR TITLE
Feat: Implement the functions in the `swap_pricing_utils` library

### PR DESCRIPTION
**Pull request by delaaxe**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

## What is the current behavior?

Resolves: #388

### CODE REVIEW

File: src/market/market_utils.cairo [hunk 1]
```diff
@@ -1030,18 +1039,50 @@ fn get_open_interest_with_pnl(
...
-/// Get the virtual inventory for swaps
-/// # Arguments
-/// * `data_store` - The data store to use.
-/// * `market` - The market address.
-/// # Returns
-/// has virtual inventory, virtual long token inventory, virtual short token inventory
+// @dev get the virtual inventory for swaps
+// @param data_store DataStore
+// @param market the market to check
+// @return returns (has virtual inventory, virtual long token inventory, virtual short token inventory)
```
Comment (Sk8erboi84) [inline src/market/market_utils.cairo:1045]: Why did you use this format for documenting

Comment (delaaxe) [reply src/market/market_utils.cairo]: was copied from solidity, fixed now

File: src/pricing/swap_pricing_utils.cairo [hunk 2]
```diff
@@ -99,16 +97,130 @@ fn get_price_impact_usd(params: GetPriceImpactUsdParams) -> i128 {
...
+    );
+
+    if price_impact_usd_for_virtual_inventory < price_impact_usd {
+        price_impact_usd_for_virtual_inventory
+    } else {
+        price_impact_usd
     }
 }
 
+/// Called by get_price_impact_usd().
```
Comment (zarboq) [inline src/pricing/swap_pricing_utils.cairo:162]: Add doc for arguments also

File: src/pricing/swap_pricing_utils.cairo [hunk 3]
```diff
@@ -99,16 +97,130 @@ fn get_price_impact_usd(params: GetPriceImpactUsdParams) -> i128 {
...
+            initial_diff_usd,
+            next_diff_usd,
+            positive_impact_factor,
+            negative_impact_factor,
+            impact_exponent_factor
+        )
+    }
+}
+
+fn get_next_pool_amount_usd(params: GetPriceImpactUsdParams) -> PoolParams {
```
Comment (zarboq) [inline src/pricing/swap_pricing_utils.cairo:213]: doc

File: src/tests/pricing/test_swap_pricing_utils.cairo [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    assert(pool_params.pool_usd_for_token_a == 101000, 'invalid');
+    assert(pool_params.pool_usd_for_token_b == 99000, 'invalid');
+    assert(pool_params.next_pool_usd_for_token_a == 101005, 'invalid');
+    assert(pool_params.next_pool_usd_for_token_b == 99004, 'invalid');
+
+    // *********************************************************************************************
+    // *                              TEARDOWN                                                     *
+    // *********************************************************************************************
+    teardown(data_store.contract_address);
+}
```
Comment (zarboq) [inline src/tests/pricing/test_swap_pricing_utils.cairo:93]: Could you add a test for `get_swap_fees` also

File: src/utils/i128.cairo [hunk 5]
```diff
@@ -90,6 +96,9 @@ impl I128Serde of Serde<i128> {
     }
 }
 
+// TODO: the next 2 functions are duplicated as calc::to_signed & calc::to_unsigned
+
+// the assert is wrong on this one
```
[the remark this answers is not in the document]
Comment (delaaxe) [reply src/utils/i128.cairo]: done directly

File: src/tests/pricing/test_swap_pricing_utils.cairo (hunk 4 above)
Comment (delaaxe) [reply src/tests/pricing/test_swap_pricing_utils.cairo:94]: added

**Review by zarboq — APPROVED** [review]

**Pull request merged**